fix mypy 1.1.1 fails

```
Run mypy --config-file .github/mypy/mypy.ini floss/ scripts/ tests/
floss/strings.py:55: error: Too many positional arguments for "StaticString"  [misc]
floss/strings.py:82: error: Too many positional arguments for "StaticString"  [misc]
floss/utils.py:234: error: Too many positional arguments for "StaticString"  [misc]
floss/stackstrings.py:53: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
floss/stackstrings.py:181: error: Too many positional arguments for "StackString"  [misc]
floss/tightstrings.py:112: error: Too many positional arguments for "TightString"  [misc]
floss/string_decoder.py:165: error: Too many positional arguments for "DecodedString"  [misc]
Found 6 errors in 5 files (checked 36 source files)
```

ref: #656

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -28,7 +28,7 @@ jobs:
     - name: Lint with black
       run: black -l 120 --check .
     - name: Check types with mypy
-      run: mypy --config-file .github/mypy/mypy.ini floss/ scripts/ tests/
+      run: mypy --config-file .github/mypy/mypy.ini --check-untyped-defs floss/ scripts/ tests/
 
   tests:
     name: Tests in ${{ matrix.python-version }} on ${{ matrix.os }}

--- a/floss/features/features.py
+++ b/floss/features/features.py
@@ -153,7 +153,7 @@ class CallsTo(Feature):
         self.locations = locations
 
     def score(self):
-        return float(self.value) / float(self.max_calls_to)
+        return float(self.value / self.max_calls_to)
 
 
 class Loop(Feature):

--- a/floss/main.py
+++ b/floss/main.py
@@ -46,7 +46,7 @@ from floss.identify import (
     find_decoding_function_features,
     get_functions_without_tightloops,
 )
-from floss.logging_ import DebugLevel
+from floss.logging_ import TRACE, DebugLevel
 from floss.stackstrings import extract_stackstrings
 from floss.tightstrings import extract_tightstrings
 from floss.string_decoder import decode_strings
@@ -263,7 +263,7 @@ def set_log_config(debug, quiet):
     if quiet:
         log_level = logging.WARNING
     elif debug >= DebugLevel.TRACE:
-        log_level = logging.TRACE
+        log_level = TRACE
     elif debug >= DebugLevel.DEFAULT:
         log_level = logging.DEBUG
     else:

--- a/floss/render/default.py
+++ b/floss/render/default.py
@@ -274,7 +274,7 @@ def get_color(color):
 
 
 def render(results, verbose, disable_headers, color):
-    sys.stdout.reconfigure(encoding="utf-8")
+    sys.__stdout__.reconfigure(encoding="utf-8")
     console = Console(file=io.StringIO(), color_system=get_color(color), highlight=False)
 
     if not disable_headers:
@@ -305,4 +305,5 @@ def render(results, verbose, disable_headers, color):
         render_heading("FLOSS DECODED STRINGS", len(results.strings.decoded_strings), console, verbose, disable_headers)
         render_decoded_strings(results.strings.decoded_strings, console, verbose, disable_headers)
 
-    return console.file.getvalue()
+    console.file.seek(0)
+    return console.file.read()

--- a/floss/results.py
+++ b/floss/results.py
@@ -190,7 +190,7 @@ class ResultDocument:
 
     @classmethod
     def parse_file(cls, path):
-        return cls.__pydantic_model__.parse_file(path)
+        return cls.parse_file(path)
 
 
 def log_result(decoded_string, verbosity):

--- a/floss/results.py
+++ b/floss/results.py
@@ -190,7 +190,8 @@ class ResultDocument:
 
     @classmethod
     def parse_file(cls, path):
-        return cls.parse_file(path)
+        # We're ignoring the following mypy error since this field is guaranteed by the Pydantic dataclass.
+        return cls.__pydantic_model__.parse_file(path)  # type: ignore
 
 
 def log_result(decoded_string, verbosity):

--- a/floss/results.py
+++ b/floss/results.py
@@ -65,6 +65,7 @@ class StackString:
     Attributes:
       function: the address of the function from which the stackstring was extracted
       string: the extracted string
+      encoding: string encoding
       program_counter: the program counter at the moment the string was extracted
       stack_pointer: the stack counter at the moment the string was extracted
       original_stack_pointer: the initial stack counter when the function was entered

--- a/floss/stackstrings.py
+++ b/floss/stackstrings.py
@@ -178,7 +178,16 @@ def extract_stackstrings(
                 )
                 for s in extract_strings(ctx.stack_memory, min_length, seen):
                     frame_offset = (ctx.init_sp - ctx.sp) - s.offset - getPointerSize(vw)
-                    ss = StackString(fva, s.string, s.encoding, ctx.pc, ctx.sp, ctx.init_sp, s.offset, frame_offset)
+                    ss = StackString(
+                        function=fva,
+                        string=s.string,
+                        encoding=s.encoding,
+                        program_counter=ctx.pc,
+                        stack_pointer=ctx.sp,
+                        original_stack_pointer=ctx.init_sp,
+                        offset=s.offset,
+                        frame_offset=frame_offset,
+                    )
                     floss.results.log_result(ss, verbosity)
                     seen.add(s.string)
                     stack_strings.append(ss)

--- a/floss/string_decoder.py
+++ b/floss/string_decoder.py
@@ -163,12 +163,12 @@ def decode_strings(
                     for delta_bytes in extract_delta_bytes(delta, ctx.decoded_at_va, fva):
                         for s in floss.utils.extract_strings(delta_bytes.bytes, min_length, seen):
                             ds = DecodedString(
-                                delta_bytes.address + s.offset,
-                                delta_bytes.address_type,
-                                s.string,
-                                s.encoding,
-                                delta_bytes.decoded_at,
-                                delta_bytes.decoding_routine,
+                                address=delta_bytes.address + s.offset,
+                                address_type=delta_bytes.address_type,
+                                string=s.string,
+                                encoding=s.encoding,
+                                decoded_at=delta_bytes.decoded_at,
+                                decoding_routine=delta_bytes.decoding_routine,
                             )
                             floss.results.log_result(ds, verbosity)
                             seen.add(ds.string)

--- a/floss/strings.py
+++ b/floss/strings.py
@@ -52,7 +52,7 @@ def extract_ascii_strings(buf, n=MIN_LENGTH) -> Iterable[StaticString]:
         reg = rb"([%s]{%d,})" % (ASCII_BYTE, n)
         r = re.compile(reg)
     for match in r.finditer(buf):
-        yield StaticString(match.group().decode("ascii"), offset=match.start(), encoding=StringEncoding.ASCII)
+        yield StaticString(string=match.group().decode("ascii"), offset=match.start(), encoding=StringEncoding.ASCII)
 
 
 def extract_unicode_strings(buf, n=MIN_LENGTH) -> Iterable[StaticString]:
@@ -79,7 +79,9 @@ def extract_unicode_strings(buf, n=MIN_LENGTH) -> Iterable[StaticString]:
         r = re.compile(reg)
     for match in r.finditer(buf):
         try:
-            yield StaticString(match.group().decode("utf-16"), offset=match.start(), encoding=StringEncoding.UTF16LE)
+            yield StaticString(
+                string=match.group().decode("utf-16"), offset=match.start(), encoding=StringEncoding.UTF16LE
+            )
         except UnicodeDecodeError:
             pass
 

--- a/floss/tightstrings.py
+++ b/floss/tightstrings.py
@@ -109,7 +109,16 @@ def extract_tightstrings(
                     logger.trace("pre_ctx strings: %s", ctx.pre_ctx_strings)
                     for s in extract_strings(ctx.stack_memory, min_length, exclude=ctx.pre_ctx_strings):
                         frame_offset = (ctx.init_sp - ctx.sp) - s.offset - floss.utils.getPointerSize(vw)
-                        ts = TightString(fva, s.string, s.encoding, ctx.pc, ctx.sp, ctx.init_sp, s.offset, frame_offset)
+                        ts = TightString(
+                            function=fva,
+                            string=s.string,
+                            encoding=s.encoding,
+                            program_counter=ctx.pc,
+                            stack_pointer=ctx.sp,
+                            original_stack_pointer=ctx.init_sp,
+                            offset=s.offset,
+                            frame_offset=frame_offset,
+                        )
                         floss.results.log_result(ts, verbosity)
                         tight_strings.append(ts)
     return tight_strings

--- a/floss/utils.py
+++ b/floss/utils.py
@@ -231,7 +231,7 @@ def extract_strings(buffer: bytes, min_length: int, exclude: Optional[Set[str]] 
         if exclude and decoded_string in exclude:
             continue
 
-        yield StaticString(decoded_string, s.offset, s.encoding)
+        yield StaticString(string=decoded_string, offset=s.offset, encoding=s.encoding)
 
 
 # FP string starts

--- a/floss/utils.py
+++ b/floss/utils.py
@@ -4,6 +4,7 @@ import time
 import inspect
 import logging
 import argparse
+import builtins
 import contextlib
 from typing import Set, Tuple, Iterable, Optional
 from collections import OrderedDict
@@ -307,10 +308,10 @@ def redirecting_print_to_tqdm():
 
     try:
         # Globaly replace print with new_print
-        inspect.builtins.print = new_print
+        builtins.print = new_print
         yield
     finally:
-        inspect.builtins.print = old_print
+        builtins.print = old_print
 
 
 @contextlib.contextmanager

--- a/scripts/idaplugin.py
+++ b/scripts/idaplugin.py
@@ -117,12 +117,13 @@ def apply_decoded_strings(decoded_strings: List[DecodedString]) -> None:
 
 
 def apply_stack_strings(
-    strings: List[Union[StackString, TightString]], lvar_cmt: bool = True, cmt: bool = True
+    stack_strings: List[StackString], tight_strings: List[TightString], lvar_cmt: bool = True, cmt: bool = True
 ) -> None:
     """
     lvar_cmt: apply stack variable comment
     cmt: apply regular comment
     """
+    strings = stack_strings + tight_strings
     for s in strings:
         if not s.string:
             continue
@@ -179,7 +180,6 @@ def main(argv=None):
         vw, selected_functions, MIN_LENGTH, verbosity=floss.render.Verbosity.VERBOSE, disable_progress=True
     )
     logger.info("decoded %d stack strings", len(stack_strings))
-    apply_stack_strings(stack_strings)
 
     logger.info("extracting tightstrings...")
     tightloop_functions = floss.identify.get_functions_with_tightloops(decoding_function_features)
@@ -191,7 +191,8 @@ def main(argv=None):
         disable_progress=True,
     )
     logger.info("decoded %d tight strings", len(tight_strings))
-    apply_stack_strings(tight_strings)
+
+    apply_stack_strings(stack_strings, tight_strings)
 
     logger.info("decoding strings...")
 

--- a/setup.py
+++ b/setup.py
@@ -63,7 +63,7 @@ setuptools.setup(
             "pycodestyle==2.10.0",
             "black==23.3.0",
             "isort==5.11.4",
-            "mypy==1.0.1",
+            "mypy==1.1.1",
             # type stubs for mypy
             "types-PyYAML==6.0.10",
             "types-tabulate==0.9.0.2",

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -9,7 +9,7 @@ import viv_utils
 import floss.main as floss_main
 import floss.stackstrings as stackstrings
 import floss.tightstrings as tightstrings
-import floss.string_decoder
+import floss.string_decoder as string_decoder
 from floss.const import MIN_STRING_LENGTH
 from floss.identify import (
     get_function_fvas,
@@ -26,18 +26,20 @@ def extract_strings(vw):
     """
     top_functions, decoding_function_features = identify_decoding_functions(vw)
 
-    for s in floss.string_decoder.decode_strings(
+    for s_decoded in string_decoder.decode_strings(
         vw, get_function_fvas(top_functions), MIN_STRING_LENGTH, disable_progress=True
     ):
-        yield s.string
+        yield s_decoded.string
 
     no_tightloop_functions = get_functions_without_tightloops(decoding_function_features)
-    for s in stackstrings.extract_stackstrings(vw, no_tightloop_functions, MIN_STRING_LENGTH, disable_progress=True):
-        yield s.string
+    for s_stack in stackstrings.extract_stackstrings(
+        vw, no_tightloop_functions, MIN_STRING_LENGTH, disable_progress=True
+    ):
+        yield s_stack.string
 
     tightloop_functions = get_functions_with_tightloops(decoding_function_features)
-    for s in tightstrings.extract_tightstrings(vw, tightloop_functions, MIN_STRING_LENGTH, disable_progress=True):
-        yield s.string
+    for s_tight in tightstrings.extract_tightstrings(vw, tightloop_functions, MIN_STRING_LENGTH, disable_progress=True):
+        yield s_tight.string
 
 
 def identify_decoding_functions(vw):
@@ -54,7 +56,7 @@ def pytest_collect_file(parent, path):
 
 class YamlFile(pytest.File):
     def collect(self):
-        spec = yaml.safe_load(self.fspath.open())
+        spec = yaml.safe_load(self.path.open())
         test_dir = os.path.dirname(str(self.fspath))
         for platform, archs in spec["Output Files"].items():
             for arch, filename in archs.items():

